cmd/lifecycle/analyzer.go

```diff
@@ -214,6 +244,15 @@ func (aa analyzeArgs) analyze() (platform.AnalyzedMetadata, error) {
 	if err != nil {
 		return platform.AnalyzedMetadata{}, cmd.FailErrCode(err, aa.platform.CodeFor(cmd.AnalyzeError), "analyzer")
 	}
+
+	if aa.runImageRef != "" {
+		ref, err := name.ParseReference(aa.runImageRef, name.WeakValidation)
+		if err != nil {
+			return platform.AnalyzedMetadata{}, cmd.FailErrCode(err, aa.platform.CodeFor(cmd.AnalyzeError), "parse reference for run image")
+		}
```

Comment:
This code is very similar to the `parseRegistry` function.
Do you think it's worth creating another helper function that we'll use both here and inside the `parseRegistry` function?

Reply:
We can't use `parseRegistry` because that returns the registry, not the reference. What would such a helper look like? I'm not sure how to reduce the duplication. It seems to me that `name.ParseReference` is doing most of the heavy lifting.